### PR TITLE
Complete the word search function in command line

### PR DESCRIPTION
**Pull request by edxu96**
- add a new click decorated search function
- change the return `DBConnector.search()` to dictionary
- some minor modifications of `DBConnector`
- didn't do setup integration

### CODE REVIEW

File: src/e2c/db_connector.py [hunk 1]
```diff
@@ -11,31 +12,59 @@
...
+@click.command()
+@click.argument("word")
+def search(word):
+    """Type in English words and return Chinese translations."""
+    res = DBConnector().search(word)
+    click.echo("\n" + res["definition"] + "\n\n" + res["trans"])
+
+
+if __name__ == "__main__":
+    search()
```
Comment (zequnyu) [inline src/e2c/db_connector.py]: This should be in a separate script file.

File: src/e2c/db_connector.py [hunk 2]
```diff
@@ -11,31 +12,59 @@
...
+                "phonetic": res[3],
+                "definition": res[4],
+                "trans": res[5],
+                "pos": res[6],
+                "collins": res[7],
+                "oxford": res[8],
+                "tag": res[9],
+                "bnc": res[10],
+                "frq": res[11],
+                "exchange": res[12],
```
Comment (zequnyu) [inline src/e2c/db_connector.py:54]: You could define a tuple/array `_key_names` outside `DBConnector` having those names and do
`return [(x, y) for x, y in zip(_key_names, res)]`

File: src/e2c/db_connector.py [hunk 3]
```diff
@@ -11,31 +12,59 @@
...
+@click.command()
+@click.argument("word")
+def search(word):
+    """Type in English words and return Chinese translations."""
+    res = DBConnector().query(word)
+    click.echo("\n" + res["definition"] + "\n\n" + res["trans"])
+
+
+if __name__ == "__main__":
+    search()
```
Comment (zequnyu) [inline src/e2c/db_connector.py:70]: This should be in another file.

File: src/e2c/db_connector.py [hunk 4]
```diff
@@ -11,31 +12,59 @@
 class DBConnector:
     """Database Connector."""
 
-    def __init__(self, path=_path):
+    def __init__(self, path: str = _path):
```
Comment (zequnyu) [inline src/e2c/db_connector.py:15]: We do not specify type hints. Instead, we write good docs to indicate this.

File: src/e2c/db_connector.py [hunk 5]
```diff
@@ -11,31 +12,59 @@
...
         """
         if pathlib.Path(path).is_file() and path.endswith(".db"):
-            self._connection = DBConnector._init_connection(path)
+            self._conn = DBConnector._init_conn(path)
         else:
             raise ValueError("Database file is missing or invalid.")
 
     @staticmethod
-    def _init_connection(path):
+    def _init_conn(path: str) -> sqlite3.Connection:
```
Comment (zequnyu) [inline src/e2c/db_connector.py:27]: No type hints.

File: src/e2c/db_connector.py [hunk 6]
```diff
@@ -11,31 +12,59 @@
...
         """Initialize database connection."""
         try:
             return sqlite3.connect(path)
         except sqlite3.Error:
             logger.exception("SQLite DB connection failed.")
 
-    def search(self, word):
+    def query(self, word: str = "wrong") -> dict:
```
Comment (zequnyu) [inline src/e2c/db_connector.py:34]: No type hints.

File: src/e2c/db_connector.py (hunk 4 above)
Comment (edxu96) [reply src/e2c/db_connector.py:15]: Why not ? It will appear automatically in docs.

File: tests/e2c/test_db_connector.py [hunk 7]
```diff
@@ -1,16 +1,23 @@
...
+    assert isinstance(res, dict) and len(res) > 0
 
 
 def test_no_file_connect():
     """Test database connection when .db file doesn't exist."""
     with pytest.raises(ValueError):
         DBConnector("./no_such_file").search("play")
+
+
+def test_search():
```
Comment (zequnyu) [inline tests/e2c/test_db_connector.py:20]: This should also be in another test file

File: src/e2c/db_connector.py (hunk 6 above)
Comment (zequnyu) [inline src/e2c/db_connector.py:34]: I think `word` should not have a default value.

File: src/e2c/db_connector.py [hunk 8]
```diff
@@ -15,27 +30,29 @@ def __init__(self, path=_path):
...
-    def _init_connection(path):
+    def _init_conn(path):
         """Initialize database connection."""
         try:
             return sqlite3.connect(path)
         except sqlite3.Error:
             logger.exception("SQLite DB connection failed.")
 
-    def search(self, word):
+    def query(self, word="wrong"):
```
Comment (zequnyu) [inline src/e2c/db_connector.py:49]: Remove the default value?

**Pull request merged**